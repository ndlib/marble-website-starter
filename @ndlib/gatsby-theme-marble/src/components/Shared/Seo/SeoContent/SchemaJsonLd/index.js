import React from 'react'
import PropTypes from 'prop-types'
import { JsonLd } from '../JsonLd'

export const SchemaJsonLd = (props) => {
  const {
    thumbnail,
    title,
    author,
    description,
    date,
    url,
    keywords,
    image,
    classification,
    creditText,
    dimensions,
    relatedLocation,
    copyrightStatus,
    publisher,
    materialType,
    physicalCharacteristic,
    language,
    acquisition,
    accessionNumber,
    campusLocation,
    identifier,
  } = props
  return (
    <JsonLd>
      {{
        '@context': 'https://schema.org',
        '@type': 'VisualArtwork',
        name: title,
        creator: [
          {
            '@type': 'Person',
            name: author,
          },
        ],
        url: url,
        locationCreated: relatedLocation,
        dateCreated: date,
        description: description,
        inLanguage: language,
        image: image,
        thumbnailUrl: thumbnail,
        keywords: keywords,
        size: dimensions || physicalCharacteristic,
        artMedium: classification,
        artform: materialType,
        creditText: creditText,
        copyrightNotice: copyrightStatus,
        publisher: publisher,
        identifier: identifier || accessionNumber,
        sponsor: acquisition,
        maintainer: campusLocation,
        provider: creditText,
      }}
    </JsonLd>
  )
}

SchemaJsonLd.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string,
  image: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
  creditText: PropTypes.string,
  classification: PropTypes.string,
  copyrightStatus: PropTypes.string,
  keywords: PropTypes.arrayOf(PropTypes.string),
  dimensions: PropTypes.string,
  relatedLocation: PropTypes.string,
  publisher: PropTypes.string,
  materialType: PropTypes.string,
  physicalCharacteristic: PropTypes.string,
  acquisition: PropTypes.string,
  accessionNumber: PropTypes.string,
  identifier: PropTypes.string,
  language: PropTypes.string,
  campusLocation: PropTypes.string,
}

export default SchemaJsonLd
