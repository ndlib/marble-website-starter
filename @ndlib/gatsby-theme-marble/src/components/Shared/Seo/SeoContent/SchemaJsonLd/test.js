import React from 'react'
import renderer from 'react-test-renderer'
import SchemaJsonLd from './'

describe('SchemaJsonLd component', () => {

  it('renders props into values', () => {
    const tree = renderer
      .create(<SchemaJsonLd
        title={'1984'}
        image={'http://fake.io/image'}
        author={'George Orwell'}
        dimensions={'2 x 4'}
        thumbnail={'http://fake.io/thumbnail'}
        url={'http://object.io/page'}
        relatedLocation={'London'}
        date={'Present'}
        description={'Dystopian nightmare'}
        language={'Newspeak'}
        keywords={'totalitarian', 'society', 'thoughtcrime', 'doublespeak'}
        physicalCharacteristic={'8in x 6in'}
        classification={'Book'}
        materialType={'Paper'}
        creditText={'BBC'}
        copyrightStatus={'Public Domain'}
        publisher={'BBC'}
        identifier={'2+2=5'}
        acquisition={'a gift from me'}
        campusLocation={'library'}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
