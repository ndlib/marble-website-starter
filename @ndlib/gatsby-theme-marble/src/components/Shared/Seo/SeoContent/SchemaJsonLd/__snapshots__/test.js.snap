// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaJsonLd component renders props into values 1`] = `null`;
